tests/unit/src/test/scala/tests/DidFocusLspSuite.scala

```diff
@@ -67,9 +70,28 @@ class DidFocusLspSuite extends BaseLspSuite("did-focus") {
       )
     } yield ()
   }
+}
+
+// https://github.com/scalameta/metals/issues/497
+class DidFocusWhileCompilingLspSuite
+    extends BaseLspSuite("did-focus-while-compiling") {
+  var fakeTime: FakeTime = _
+  val compileDelayMillis = 5000
+  override def time: Time = fakeTime
+
+  // sleep 10s during the compilation, so that we can make sure
+  // invoking `didFocus` during compilation.
+  override def beforeEach(context: BeforeEach): Unit = {
+    fakeTime = new FakeTime()
+    onStartCompilation = () => {
+      Thread.sleep(compileDelayMillis)
+    }
+    super.beforeEach(context)
+  }
 
-  // Ignore flaky test, see the details: https://github.com/scalameta/metals/pull/3752#issuecomment-1079878023
-  test("497".ignore) {
+  test(
+    "Trigger compilation by didFocus when current compile may affect focused buffer"
```

Comment:
I think we have the naming convention to use a shorter name written in kebab-case. Current name could be extracted as a comment I think